src/server/game/Server/Packets/MiscPackets.h

```diff
@@ -285,6 +285,18 @@ namespace WorldPackets
             ObjectGuid Roller;
         };
 
+        class TogglePvP final : public ClientPacket
+        {
+        public:
+            TogglePvP(WorldPacket&& packet) : ClientPacket(CMSG_TOGGLE_PVP, std::move(packet)) { }
+
+            void Read() override;
+
+            bool HasPvPStatus() const { return GetSize() == 1; }
```

Comment:
Don't like this way to check, if packet has data. I think we need generic way for clientpacket class

Reply:
Enable could be Optional<bool>

Reply:
I agree with Optional but not with operator>> suggestion, it is better to leave initializing it decoupled with reading the field

Reply:
><%Shauren> its probably easier to just slap a getsize check on it with current state of bytebuffer :D
<%jackpoz> what about a CanRead<T>() ?
return _rpos + sizeof(T) <= size();
<%Shauren> if you want to add it, make another overload that accepts (size_t pos) too for parity with read overloads
but sounds good

Reply:
I think, it should be done outside the pr, and after I will update this PR using Optional for Enable flag